python/tvm/relay/op/_tensor_grad.py

```diff
@@ -511,7 +511,7 @@ def batch_matmul_grad(orig, grad):
 @register_gradient("reshape")
 def reshape_grad(orig, grad):
     """Gradient of reshape"""
-    return [reshape_like(grad, orig.args[0]), orig.args[1]]
+    return [reshape_like(grad, orig.args[0])]
```

Comment:
Do we need to register a gradient op to `dyn.reshape`?

Reply:
The numerical approx_grad doesn't seem to work for this case, and it doesn't seem the dynamic version was tested before. I need to look at the autodiff infrastructure more closely, perhaps this should be a second PR?

Reply:
Agree.